runtime/src/lib.rs

```diff
@@ -727,6 +730,52 @@ impl pallet_preimage::Config for Runtime {
     type ByteDeposit = PreimageByteDeposit;
 }
 
+impl InstanceFilter<Call> for ProxyType {
+    fn filter(&self, c: &Call) -> bool {
+        match self {
+            ProxyType::Any => true,
+            ProxyType::NonTransfer => matches!(c, Call::Balances(..)),
```

Comment:
Expanding on 2., we should also block all types of creating/trading on markets. For example, you could "transfer" money from Alice to Bob by doing this:

- Alice opens a market with an obvious question and provides liquidity
- Bob buys all the correct tokens and redeems them later

So even if transfers aren't allowed, you can still steal funds if you get a hold of someone's proxy.

I wonder if it's easier to have proxies for specific purposes, and not for _excluding_ purposes. So, for example, one proxy for democracy, one for trading/creating markets, etc.

Reply:
Acala is going in this direction: https://github.com/AcalaNetwork/Acala/blob/9f11fa6f6635e7c09b134b515b3f4e8713b2cdc0/runtime/acala/src/lib.rs#L1262-L1317

Reply:
> I wonder if it's easier to have proxies for specific purposes, and not for excluding purposes. So, for example, one proxy for democracy, one for trading/creating markets, etc.

I would say this is a good approach.

For now I opted to remove `NonTransfer`. We can reconsider adding `NonTransfer` back later or continue to add use-case specific proxies (probably the better approach) in future changes. 

Also added `CancelProxy` from the Acala example because that does seem useful to me for the initial proxy configurations I want to make.

Reply:
Changes made in fe87bf75dcacaac94e4c14590cefa582f8953263